File: LlmIntegration/Util/LlmModel.cs
using System.Text.Json.Serialization;

namespace LLMIntegration.Util;

public record LlmModel(
    [property: JsonPropertyName("provider")] LlmProvider Provider,
    [property: JsonPropertyName("modelName")] string ModelName,
    [property: JsonPropertyName("modelDescription")] string ModelDescription,
    [property: JsonPropertyName("modelIdentifier")] string ModelIdentifier,
    [property: JsonPropertyName("maxCompletionTokens")] int MaxCompletionTokens = 2048,
    [property: JsonPropertyName("legacy")] bool IsLegacy = false);

File: Frontend/src/model/conversation/llmModel.ts
export type LlmProvider = "Anthropic"|"Google"|"OpenAi"|"X";

export type LlmModel = {
    provider: LlmProvider,
    modelName: string,
    modelDescription: string,
    modelIdentifier: string
} 

export type LlmProviderGroup = {
    provider: LlmProvider,
    models: LlmModel[];
}